test/integration/coredump/test_coredump.py

```diff
@@ -19,31 +19,42 @@
 # For the cases above generated files are removed after tests but in general
 # there is no guarantee. So tests that use this fixture are disabled by default
 # (marked with skipif) in order to avoid coredumps "leaks".
-# One may launch them explicitly using TT_ENABLE_COREDUMP_FIXTURE environment
-# TT_ENABLE_COREDUMP_FIXTURE=1 python3 -m pytest test/integration/coredump/.
+# One may launch them explicitly using TT_ENABLE_COREDUMP_TESTS environment
+# TT_ENABLE_COREDUMP_TESTS=1 python3 -m pytest test/integration/coredump/.
 
 
-@pytest.fixture(scope="session")
-def coredump(tmp_path_factory) -> Path:
-    coredump_tmpdir = tmp_path_factory.mktemp("coredump")
+skip_coredump_cond = os.getenv('TT_ENABLE_COREDUMP_TESTS') is None
+skip_coredump_reason = "Should be launched explicitly to control coredump it produces"
+
+
+def generate_coredump(tmp_path_factory, env=None) -> Path:
+    coredump_dir = tmp_path_factory.mktemp("coredump")
     with open('/proc/sys/kernel/core_pattern', 'r') as f:
-        core_pattern = f.read()
+        core_pattern = f.read().strip()
+
+    def coredump_apport(core_source, outdir):
+        rc, _ = run_command_and_get_output(['apport-unpack', core_source, outdir])
+        assert rc == 0
+        return outdir / 'CoreDump'
 
-    def apport_crash_to_coredump(crash):
-        apport_unpack_dir = coredump_tmpdir / 'apport-unpack'
-        rc, output = run_command_and_get_output(['apport-unpack', crash, apport_unpack_dir])
-        return apport_unpack_dir / 'CoreDump'
+    def coredump_systemd(core_source, outdir):
+        core = outdir / core_source.stem
+        rc, _ = run_command_and_get_output(['coredumpctl', 'dump', f'--output={core}'])
```

Comment:
If you don't need `output` why you keep the same function here?
```suggestion
        rc = subprocess.run(['coredumpctl', 'dump', f'--output={core}']).returncode
```
This way the output of the command will not be captured and deleted, which will then appear in the log in case of problems with the test, and this in itself can help in finding problems.